### DISCUSSION TITLE
Ref Block Diagram Generation

### DISCUSSION THREAD

**Discussion by MichaelStarks** [nasa/fprime#1747] [Q&A] [answered]
What tools were used to make the ISFApp.pdf found in Ref/Top?

**Comment by timcanham** [comment, marked as answer]
We used MagicDraw, a UML modeling tool, in earlier versions of F Prime. That tool has been deprecated and the diagram is out of date, so we will remove it in the next release.

**Reply by MichaelStarks** [reply]
Is there a way to generate a full system diagram with the current tools? Following the instructions in the tutorial only generates a diagram showing the direct connection between components.

**Comment by SMorettini** [comment]
Did you tried this tool https://github.com/fprime-community/fprime-layout ?
